packages/d3fc-webgl/src/shaders/line/baseShader.js

```diff
@@ -11,11 +11,21 @@ export default () => {
         .appendHeader(`attribute float aXValue; attribute float aYValue;`) // the current vertex positions
         .appendHeader(`attribute float aNextXValue; attribute float aNextYValue;`) // the next vertex positions
         .appendHeader(`attribute float aPrevXValue; attribute float aPrevYValue;`) // the previous vertex positions
+        .appendHeader(`attribute float aDefined;`)
+        .appendHeader(`varying float vDefined;`)
+        .appendBody(`vDefined = aDefined;`)
         .appendBody(`vec4 curr = vec4(aXValue, aYValue, 0, 1);`)
         .appendBody(`gl_Position = curr;`)
         .appendBody(`vec4 next = vec4(aNextXValue, aNextYValue, 0, 0);`)
         .appendBody(`vec4 prev = vec4(aPrevXValue, aPrevYValue, 0, 0);`);
 
+    fragmentShader
+        .appendHeader(`varying float vDefined;`)
+        .appendBody(`if (vDefined < 0.5) {
+            discard;
```

Comment:
I think discard is a control flow statement in and of itself, don't think the return is required.